### ISSUE TITLE
Template format error: Amplify add auth

### ISSUE THREAD

**Issue by royjit** [aws-amplify/amplify-cli#6237]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
Amplify auth with social federation end up in error.

**Amplify CLI Version**
4.36.0

**To Reproduce**
1.  Follow these step in CLI
```console

amplify add auth
Using service: Cognito, provided by: awscloudformation

 The current configured provider is Amazon Cognito.

 Do you want to use the default authentication and security configuration? Default configuration with Social Provider (Federation)
 Warning: you will not be able to edit these selections.
 How do you want users to be able to sign in? Username
 Do you want to configure advanced settings? No, I am done.
 What domain name prefix do you want to use? <xx>
 Enter your redirect signin URI: myapp://
? Do you want to add another redirect signin URI No
 Enter your redirect signout URI: myapp://
? Do you want to add another redirect signout URI No
 Select the social providers you want to configure for your user pool:
Successfully added auth resource <xx> locally
```
2. Push the newly created auth `amplify push`. 
3. Amplify push fail with the following error
```console
Following resources failed

Resource Name: amplify-xxx-deployment (AWS::S3::Bucket)
Event Type: update
Reason: Resource update cancelled
URL: ...


Resource Name: amplify-xxx-authRole-idp (AWS::IAM::Role)
Event Type: create
Reason: Resource creation cancelled


✖ An error occurred when pushing the resources to the cloud

Resource is not in the state stackUpdateComplete
An error occurred during the push operation: Resource is not in the state stackUpdateComplete
```
4. Cloud formation has the following error
```console
Template format error: Unresolved resource dependencies [hostedUIProviderMeta] in the Resources block of the template
```
**Expected behavior**
`amplify push` should work without any error

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Mac/Windows/Ubuntu] Mac
 - Node Version. v12.18.3

**Additional context**

**Comment by akshbhu** [comment]
Hi @royjit 

We have released a fix for this in v4.40. Can you try with the latest version v4.40.1 ?
let me know if this works for you?

**Comment by royjit** [comment]
@akshbhu 

Updating the cli fixed the issue. Closing this.
